### ISSUE TITLE
PowerShell Integrated Console prompt does not appear

### ISSUE THREAD

**Issue by Liturgist** [PowerShell/vscode-powershell#2719]
PowerShell Integrated Console

After installing the Microsoft PowerShell Extension, it appears that I can choose a TERMINAL of "powershell" or "PowerShell Integrated Console." 

The "powershell" TERMINAL appears to work as expected, but no debugging can be done.

"""Cannot debug or run a PowerShell script until the PowerShell session has started. Wait for the PowerShell session to finish starting and try again.
"""

The "PowerShell Integrated Console" TERMINAL does not. It appears hung. It accepts no keyboard entry.

This is based on a fresh install of VS Code from VSCodeUserSetup-x64-1.45.1.exe.

```
Version: 1.45.1 (user setup)
Commit: 5763d909d5f12fe19f215cbfdd29a91c0fa9208a
Date: 2020-05-14T08:27:35.169Z
Electron: 7.2.4
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363
```

```
PowerShell extension 2020.4.0
%USERPROFILE%\.vscode\extensions\ms-vscode.powershell-1.8.2\ms-vscode.powershell-1.8.2
    (PS: There is no "examples" subdirectory)
```

**Comment by ninmonkey** [comment]
The debugger and extension are the 'integrated terminal' The other one is a regular shell.
It doesn't boot up instantly. Hit f5, it'll say it's loading, then you'll see it finish. Then hit f5 to use it. After the first load it stays loaded.

**Comment by Liturgist** [comment]
When VS Code is first started, I get the Welcome doc and a regular shell PowerShell.

Pressing F5:

- brings up the the "Select Environment" list,
- starts a "PowerShell Integrated Console" terminal,
- and displays "Starting PowerShell..." in yellow on the bottom status line.

The PowerShell Integrated Console terminal displays "=====> PowerShell Integrated Console <=====" and nothing else.

Clicking on "Starting PowerShell..." brings up a notification:
```
Running the contributed command:
'PowerShell.Show.SessionMenu' failed.
```
How can I tell which PowerShell executable is being used? I have two (2) in the PATH on my system.

```
C:>where powershell
C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe

C:>where pwsh
C:\Program Files\PowerShell\7.0.1\7\pwsh.exe
```

**Comment by Liturgist** [comment]
It has been "Starting PowerShell..." for over thirty (30) minutes. What should I expect?

**Comment by TylerLeonhardt** [comment]
Thanks for the info! Can you also [attach the logs](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md#providing-information-about-your-environment) here so we can get a better idea of what's going on?

**Comment by TylerLeonhardt** [comment]
Also is this machine locked down in any way?

**Comment by Liturgist** [comment]
Logs are attached from a recent run. I did not find any text about "error" or "fail" in them.

[vscode-powershell.log](https://github.com/PowerShell/vscode-powershell/files/4674337/vscode-powershell.log)
[EditorServices.log](https://github.com/PowerShell/vscode-powershell/files/4674338/EditorServices.log)
[StartEditorServices.log](https://github.com/PowerShell/vscode-powershell/files/4674339/StartEditorServices.log)

**Comment by y-halim** [comment]
I have the same issue with the same vesion of VS Code that I installed 3 days ago.

Version: 1.45.1 (user setup)
Commit: 5763d909d5f12fe19f215cbfdd29a91c0fa9208a
Date: 2020-05-14T08:27:35.169Z
Electron: 7.2.4
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18362

[vscode-powershell.log](https://github.com/PowerShell/vscode-powershell/files/4681586/vscode-powershell.log)
[EditorServices.log](https://github.com/PowerShell/vscode-powershell/files/4681587/EditorServices.log)
[StartEditorServices.log](https://github.com/PowerShell/vscode-powershell/files/4681588/StartEditorServices.log)

**Comment by TylerLeonhardt** [comment]
Does this machine use Avecto or BeyondSoft at all?

Is there any enforcement via Group Policy on this machine?

Is there any anti-virus on the machine?

Have you configured the machine in any bizarre ways... That might be non-traditional?

**Comment by Liturgist** [comment]
Does this machine use Avecto or BeyondSoft at all? **No**
Is there any enforcement via Group Policy on this machine? **`gpresult /Scope User /v` does not report any**
Is there any anti-virus on the machine? **Yes, McAfee**
Have you configured the machine in any bizarre ways... That might be non-traditional? **No**

**Comment by Liturgist** [comment]
Requested Repro-Info provided.

**Comment by SydneyhSmith** [comment]
Thanks so much for all the information, we are still having some trouble diagnosing the crash...it would be really helpful if you could also attach the payloadlogs--instructions for how to do that are [here ](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md#provide-language-server-protocol-payload-logs)

**Comment by Liturgist** [comment]
@SydneyhSmith  - Please let me know if I can provide anything else. Thanks.

There is no "PowerShell Editor Services" item in the drop-down listbox. I chose "PowerShell Extension Logs."

```
5/26/2020 7:24:41 PM [NORMAL] - Visual Studio Code v1.45.1 64-bit
5/26/2020 7:24:41 PM [NORMAL] - PowerShell Extension v2020.4.0
5/26/2020 7:24:41 PM [NORMAL] - Operating System: Windows 64-bit
5/26/2020 7:24:41 PM [NORMAL] - Language server starting --
5/26/2020 7:24:41 PM [NORMAL] -     PowerShell executable: C:\Program Files\PowerShell\7\pwsh.exe
5/26/2020 7:24:41 PM [NORMAL] -     PowerShell args: -NoProfile -NonInteractive -ExecutionPolicy Bypass -Command Import-Module 'c:\Users\pwatson\.vscode\extensions\ms-vscode.powershell-2020.4.0\modules\PowerShellEditorServices\PowerShellEditorServices.psd1'; Start-EditorServices -HostName 'Visual Studio Code Host' -HostProfileId 'Microsoft.VSCode' -HostVersion '2020.4.0' -AdditionalModules @('PowerShellEditorServices.VSCode') -BundledModulesPath 'c:\Users\pwatson\.vscode\extensions\ms-vscode.powershell-2020.4.0\modules' -EnableConsoleRepl -LogLevel 'Verbose' -LogPath 'c:\Users\pwatson\.vscode\extensions\ms-vscode.powershell-2020.4.0\logs\1590539081-e61e7b25-a838-4c9c-baf4-ba1792417ad11590539066624\EditorServices.log' -SessionDetailsPath 'c:\Users\pwatson\.vscode\extensions\ms-vscode.powershell-2020.4.0\sessions\PSES-VSCode-18956-257991' -FeatureFlags @() 
5/26/2020 7:24:41 PM [NORMAL] -     PowerShell Editor Services args: Import-Module 'c:\Users\pwatson\.vscode\extensions\ms-vscode.powershell-2020.4.0\modules\PowerShellEditorServices\PowerShellEditorServices.psd1'; Start-EditorServices -HostName 'Visual Studio Code Host' -HostProfileId 'Microsoft.VSCode' -HostVersion '2020.4.0' -AdditionalModules @('PowerShellEditorServices.VSCode') -BundledModulesPath 'c:\Users\pwatson\.vscode\extensions\ms-vscode.powershell-2020.4.0\modules' -EnableConsoleRepl -LogLevel 'Verbose' -LogPath 'c:\Users\pwatson\.vscode\extensions\ms-vscode.powershell-2020.4.0\logs\1590539081-e61e7b25-a838-4c9c-baf4-ba1792417ad11590539066624\EditorServices.log' -SessionDetailsPath 'c:\Users\pwatson\.vscode\extensions\ms-vscode.powershell-2020.4.0\sessions\PSES-VSCode-18956-257991' -FeatureFlags @() 
5/26/2020 7:24:41 PM [NORMAL] - pwsh.exe started.
5/26/2020 7:24:41 PM [NORMAL] - Waiting for session file
5/26/2020 7:24:43 PM [NORMAL] - Session file found
5/26/2020 7:24:43 PM [NORMAL] - Registering terminal close callback
5/26/2020 7:24:43 PM [NORMAL] - Registering terminal PID log callback
5/26/2020 7:24:43 PM [NORMAL] - pwsh.exe PID: 4444
5/26/2020 7:24:43 PM [NORMAL] - Language server started.
5/26/2020 7:24:43 PM [NORMAL] - {"status":"started","languageServiceTransport":"NamedPipe","languageServicePipeName":"\\\\.\\pipe\\PSES_sdyh4nmk.fh2","debugServiceTransport":"NamedPipe","debugServicePipeName":"\\\\.\\pipe\\PSES_cc1qu3nv.wnp"}
5/26/2020 7:24:43 PM [NORMAL] - Connecting to language service on pipe \\.\pipe\PSES_sdyh4nmk.fh2...
5/26/2020 7:24:43 PM [NORMAL] - Language service connected.
```
Output from PowerShell Integrated Console

```
VERBOSE: Removing PSReadLine
VERBOSE: Adding AssemblyResolve event handler for new AssemblyLoadContext dependency loading
VERBOSE: Loading EditorServices
VERBOSE: 
== Build Details ==
- Editor Services version: 2.1.0
- Build origin:            Release
- Build time:              2020-04-15 21:53:56       

VERBOSE: 
== Host Startup Configuration Details ==
 - Host name:                 Visual Studio Code Host
 - Host version:              2020.4.0
 - Host profile ID:           Microsoft.VSCode       
 - PowerShell host type:      System.Management.Automation.Internal.Host.InternalHost

 - REPL setting:              PSReadLine
 - Session details path:      c:\Users\pwatson\.vscode\extensions\ms-vscode.powershell-2020.4.0\sessions\PSES-VSCode-18956-257991
 - Bundled modules path:      c:\Users\pwatson\.vscode\extensions\ms-vscode.powershell-2020.4.0\modules
 - Additional modules:        PowerShellEditorServices.VSCode
 - Feature flags:

 - Log path:                  c:\Users\pwatson\.vscode\extensions\ms-vscode.powershell-2020.4.0\logs\1590539081-e61e7b25-a838-4c9c-baf4-ba1792417ad11590539066624\EditorServices.log
 - Minimum log level:         Verbose

 - Profile paths:
   + AllUsersAllHosts:       C:\Program Files\PowerShell\7\profile.ps1
   + AllUsersCurrentHost:    C:\Program Files\PowerShell\7\Microsoft.VSCode_profile.ps1
   + CurrentUserAllHosts:    C:\Users\pwatson\Documents\PowerShell\profile.ps1
   + CurrentUserCurrentHost: C:\Users\pwatson\Documents\PowerShell\Microsoft.VSCode_profile.ps1

VERBOSE: 
== Console Details ==
 - Console input encoding: OEM United States
 - Console output encoding: OEM United States
 - PowerShell output encoding: Unicode (UTF-8)

VERBOSE: 
== PowerShell Details ==
- PowerShell version: 7.0.0
- Language mode:      FullLanguage

VERBOSE: 
== Environment Details ==
 - OS description:  Microsoft Windows 10.0.18363
 - OS architecture: X64
 - Process bitness: 64

VERBOSE: Updated PSModulePath to: 'C:\Users\pwatson\Documents\PowerShell\Modules;C:\Program Files\PowerShell\Modules;c:\program files\powershell\7\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules;c:\Users\pwatson\.vscode\extensions\ms-vscode.powershell-2020.4.0\modules'
VERBOSE: Loading PowerShell Editor Services
VERBOSE: Loading PSES DLL using new assembly load context
VERBOSE: Starting EditorServices
VERBOSE: Starting server, deregistering host logger and registering shutdown listener


                  =====> PowerShell Integrated Console <=====
```

**Comment by y-halim** [comment]
The problem disappeared after I restarted my computer.

**Comment by Liturgist** [comment]
I have restarted my computer many times and it continues to fail in the same way.

**Comment by SydneyhSmith** [comment]
Thanks @Liturgist for the additional information, unfortunately your logs do not look quite as expected, would you mind sharing what settings you have enabled? Thanks again!

**Comment by Liturgist** [comment]
> 
> 
> Thanks @Liturgist for the additional information, unfortunately your logs do not look quite as expected, would you mind sharing what settings you have enabled? Thanks again!

@SydneyhSmith - I would be happy to share the settings. How do I generate a list of the settings? Is it a .json file? If so, where?

**Comment by TylerLeonhardt** [comment]
@Liturgist you'll wanna grab the file from here:

https://code.visualstudio.com/docs/getstarted/settings#_settings-file-locations

**Comment by Liturgist** [comment]
Here is the content of "%APPDATA%\Code\User\settings.json". There are a few .json files under "%USERPROFILE%\.vscode", but the all appear to be related to azure or Plaster.

```
C:>type "%APPDATA%\Code\User\settings.json"
{
    "window.zoomLevel": 2
    ,"terminal.integrated.shell.windows":
        "C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\powershell.exe",
    "workbench.activityBar.visible": true,
    "powershell.developer.powerShellExeIsWindowsDevBuild": true,
    "team.showWelcomeMessage": false,
    "[powershell]": {},
    "files.associations": {
        "*.ps1": "powershell"
    },
    "powershell.codeFormatting.pipelineIndentationStyle": "IncreaseIndentationAfterEveryPipeline",
    "powershell.developer.editorServicesLogLevel": "Verbose"
}
```

**Comment by TylerLeonhardt** [comment]
@Liturgist if you remove these settings:
```
    "powershell.developer.powerShellExeIsWindowsDevBuild": true,
    "[powershell]": {},
    "powershell.developer.editorServicesLogLevel": "Verbose"
```

does that help at all?

**Comment by TylerLeonhardt** [comment]
If not, can you add:
```
"powershell.developer.editorServicesLogLevel": "Diagnostic"
```
and give us the logs?

**Comment by Liturgist** [comment]
@TylerLeonhardt - Removing the three (3) lines produced the same failure. I will attach the log files. The result is still:

```
                  =====> PowerShell Integrated Console <=====

<hang>
```
[EditorServices.log](https://github.com/PowerShell/vscode-powershell/files/4706270/EditorServices.log)
[StartEditorServices.log](https://github.com/PowerShell/vscode-powershell/files/4706271/StartEditorServices.log)
[vscode-powershell.log](https://github.com/PowerShell/vscode-powershell/files/4706272/vscode-powershell.log)

**Comment by peter-sahajian** [comment]
Hello everyone, I just want to point something out as I am observing the same issues with the same error popup and symptoms, and one thing I want to call out is that it occurs when opening/having a Powershell (.ps1) file open. It seems to trigger the Powershell extension, and you will notice in the reproduced steps (and my image attached) that the bottom right indicates VS Code is "Starting Powershell"... this is hanging.
When I personally use the command line to uninstall Powershell extension: `ms-vscode.powershell`, the hanging issue goes away.

This is no matter which version of VS Code I install, or if the ms-vscode.powershell extension is up to date (I had 2020.4.0).

I don't know if they are related but I wanted to call out my similar experience.

**EDIT** After the Starting Powershell eventually times out/errors, I get control of VS code back.
Updated with more community guideline bug report info:

| Q        | A|
| ------------- |:-------------:|
| Version of VS Code     | `1.44.2` (system) and `1.45.1` (system) |
| Your operating system     | Windows 10      |
| List of extensions that you have installed|eamodio.gitlens|
||mauve.terraform|
||ms-dotnettools.csharp|
||ms-python.python|
||**ms-vscode.powershell**|
||redhat.vscode-yaml|
||Tyriar.shell-launcher|
|Reproducible steps|Have ms-vscode.powershell installed as extension|
||Open vs code ~with a .ps1 file in editor~|
||Wait for error message to pop up|
|Expected vs. saw|VS Code should open the file with no issues|

Eventually the VS Code Powershell startup fails, and this is what I see in the logs:

6/1/2020 4:08:17 PM [NORMAL] - Visual Studio Code v1.44.2 64-bit
6/1/2020 4:08:17 PM [NORMAL] - PowerShell Extension v2020.4.0
6/1/2020 4:08:17 PM [NORMAL] - Operating System: Windows 64-bit
6/1/2020 4:08:17 PM [NORMAL] - Language server starting --
6/1/2020 4:08:17 PM [NORMAL] -     PowerShell executable: C:\windows\System32\WindowsPowerShell\v1.0\powershell.exe
6/1/2020 4:08:17 PM [NORMAL] -     PowerShell args: <omitted>
6/1/2020 4:08:17 PM [NORMAL] -     PowerShell Editor Services args: <omitted>
6/1/2020 4:08:17 PM [NORMAL] - powershell.exe started.
6/1/2020 4:08:17 PM [NORMAL] - Waiting for session file
6/1/2020 4:10:17 PM [NORMAL] - Error occurred retrieving session file:
Timed out waiting for session file to appear.
6/1/2020 4:10:17 PM [NORMAL] - Language server startup failed.
6/1/2020 4:10:17 PM [ERROR] - The language service could not be started: 
6/1/2020 4:10:17 PM [ERROR] - Timed out waiting for session file to appear.

Images
![image](https://user-images.githubusercontent.com/11936329/83530502-d05ecc00-a4b9-11ea-985f-00ec9f46d854.png)

**Note**: this has been discussed a bit in the vscode repo, found here https://github.com/microsoft/vscode/issues/98822

**Comment by Liturgist** [comment]
I have the same result with Preview version.
```
=====> PowerShell Preview Integrated Console v2020.6.0 <=====
<hang>
```

**Comment by SydneyhSmith** [comment]
Thanks @Liturgist for all the information! 

Thanks also @peter-sahajian for the detailed repro steps it would be really helpful if you could also attach the logs--instructions for how to do that are [here ](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md#providing-information-about-your-environment)

**Comment by Liturgist** [comment]
I changed the setting as you specified, @SydneyhSmith. Although, I do not know how it knew to find powershell.exe 5.1. This produced:

`You have an older version of PackageManagement known to cause issues with the PowerShell extension. Would you like to update PackageManagement (You will need to restart the PowerShell extension after)?
`

followed by:

`PackageManagement updated, If you already had PackageManagement loaded in your session, please restart the PowerShell extension.`

=== Perhaps there is some improvement. Starting VS code with a .ps1 file produces:
    (I have the version and profile script location emitted in the profile scripts)

```
                  =====> PowerShell Integrated Console <=====

5.1.18362.752 AllUsersAllHosts @ C:\Windows\System32\WindowsPowerShell\v1.0\profile.ps1
5.1.18362.752 CurrentUserAllHosts @ C:\Users\pwatson\Documents\WindowsPowerShell\profile.ps1
PS C:\src\vscode-powershell> 
```
===

Pressing F5 to start debugging produces:
    (in Terminal "Task - BuildAll (vscode-powershell)")

```
> Executing task in folder vscode-powershell: Invoke-Build BuildAll <

Invoke-Build: The term 'Invoke-Build' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
The terminal process terminated with exit code: 1

Terminal will be reused by tasks, press any key to close it.
```

**Comment by TylerLeonhardt** [comment]
@Liturgist 

```
PS C:\src\vscode-powershell >
```

Did you clone this repo or something? It appears you have the extension's source.

**Comment by Liturgist** [comment]
@TylerLeonhardt, yes I cloned the PowerShell extension. Is that a source of problems?

**Comment by peter-sahajian** [comment]
> Thanks @Liturgist for all the information, sorry to hear that you are still hitting so many issues, it would be great if you could try setting your default version to Windows PowerShell 5.1 and see if you are able to start. The setting to do that is:
> 
> ```
>     "powershell.powerShellDefaultVersion": "Windows PowerShell (x64)",
> ```
> 
> Thanks also @peter-sahajian for the detailed repro steps it would be really helpful if you could also attach the logs--instructions for how to do that are [here ](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md#providing-information-about-your-environment)

@SydneyhSmith I have set debug level to Diagnostic and attached a zip of three files that appear in the Logs dir of the extension - ignore the "- Copy" in each of the filenames, it was the easiest to prepare for sharing here.

I produced these logs simply by:
1. opening the settings and changing to Diagnostic,
2. re-opening the vs code editor
3. Opening the extension page for ms-vscode.powershell (which triggers the Powershell Integrated Console and the "Starting Powershell" issue.

In my situation, it would seem there is some issue in finding/loading the session

[vscode-powershell.zip](https://github.com/PowerShell/vscode-powershell/files/4720380/vscode-powershell.zip)

**Comment by SydneyhSmith** [comment]
@peter-sahajian thanks for the additional info, were you also able to capture payload logs? 
--instructions for how to do that are [here ](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md#provide-language-server-protocol-payload-logs)
Thanks again!

**Comment by TylerLeonhardt** [comment]
@Liturgist is there a reason why you cloned the PowerShell extension? You shouldn't need that at all unless you're trying to work on the extension itself. What you're seeing when you press F5 is debug configuration for the PowerShell extension.

**Comment by peter-sahajian** [comment]
> @peter-sahajian thanks for the additional info, were you also able to capture payload logs? 
> --instructions for how to do that are [here ](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md#provide-language-server-protocol-payload-logs)
> Thanks again!

@SydneyhSmith I have not had a chance to collect those, as I had to uninstall in order to unblock my workload. I will try to add those today. Do the other logs add any insight?

**Comment by Liturgist** [comment]
> @Liturgist is there a reason why you cloned the PowerShell extension? You shouldn't need that at all unless you're trying to work on the extension itself. What you're seeing when you press F5 is debug configuration for the PowerShell extension.

@TylerLeonhardt, I only cloned it to be able to see the code. I did not clone it thinking that it was needed to make it work.

**Comment by peter-sahajian** [comment]
Hi @SydneyhSmith , I do not get the option in the OUTPUT dropdown to select Powershell Editor Service logs. With the following JSON settings in VSCode,

```
{
    "powershell.developer.editorServicesLogLevel": "Diagnostic",
    "powershell editor services.trace.server":"verbose"
}
```

This is what I can see in my dropdown for Output.

![image](https://user-images.githubusercontent.com/11936329/83881756-21b3c900-a70f-11ea-87c5-cb2b7fcd9542.png)

**Comment by ninmonkey** [comment]
@peter-sahajian 

It's a lot easier to view, copy, and search logs using the vscode command `powershell open powershell extension logs folder`  If you type `open log` it will let you view the other 2 log folders as well. 

![image](https://user-images.githubusercontent.com/3892031/84040237-fd9ef480-a967-11ea-9bb6-fdea44477dad.png)

It adds syntax highlighting. 

Search using the regex `\[err\]|exception`

![image](https://user-images.githubusercontent.com/3892031/84041554-95e9a900-a969-11ea-862a-a6631b4b91dc.png)

If it's open you can `right click` and it will jump to the actual folder -- making it easier to copy. 

![image](https://user-images.githubusercontent.com/3892031/84041840-ed881480-a969-11ea-98b5-53b7590af74f.png)

**Comment by ArtisanByteCrafter** [comment]
I'm having this issue as well today across two computers.

It loads powershell 7.0 just fine, but refuses to load Windows Powershell 5.1 and hangs indefinitely. All the above logs look normal, and i've tried the insiders and stable of both vscode and the ps extenstion, all to no avail.

Just in case, here are the logs.
[EditorServices.log](https://github.com/PowerShell/vscode-powershell/files/4766276/EditorServices.log)
[StartEditorServices.log](https://github.com/PowerShell/vscode-powershell/files/4766277/StartEditorServices.log)
[vscode-powershell.log](https://github.com/PowerShell/vscode-powershell/files/4766278/vscode-powershell.log)

**Comment by ArtisanByteCrafter** [comment]
I seem to have pinpointed the exact release that started this behavior for me.

I downgraded every version of the PS Preview extension until the issue was fixed. I had to go back to `v2020.1.0` for the issue to finally become resolved. Once i updated to `2020.2.0` or anything more recent, it breaks again. This might be useful for finding the bug, and for anyone who needs a working version in the mean time.

**Comment by SydneyhSmith** [comment]
Thanks @ArtisanByteCrafter for all the information, and sorry to hear that you are hitting this issue...are you using Avecto/BeyondSoft?

**Comment by ArtisanByteCrafter** [comment]
> Avecto/BeyondSoft

I'm not, no

**Comment by olvd** [comment]
> Does this machine use Avecto or BeyondSoft at all?
> 
> Is there any enforcement via Group Policy on this machine?
> 
> Is there any anti-virus on the machine?
> 
> Have you configured the machine in any bizarre ways... That might be non-traditional?

Hi, Can you expand on Avecto?
That is my case, experiencing the same issue.
Thanks.

**Comment by badgerbaj** [comment]
After Updating VS Code today, I am experiencing this issue with the 2020.6.0 build of PowerShell extension. Downgrading to 2020.4.0 fixes the problem for me. 

VS Code details:
Version: 1.46.0 (system setup)
Commit: a5d1cc28bb5da32ec67e86cc50f84c67cc690321
Date: 2020-06-10T09:03:20.462Z
Electron: 7.3.1
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.17763

**Comment by peter-sahajian** [comment]
> After Updating VS Code today, I am experiencing this issue with the 2020.6.0 build of PowerShell extension. Downgrading to 2020.4.0 fixes the problem for me. 
> 
> VS Code details:
> Version: 1.46.0 (system setup)
> Commit: a5d1cc28bb5da32ec67e86cc50f84c67cc690321
> Date: 2020-06-10T09:03:20.462Z
> Electron: 7.3.1
> Chrome: 78.0.3904.130
> Node.js: 12.8.1
> V8: 7.8.279.23-electron.0
> OS: Windows_NT x64 10.0.17763
> 

For some reason I was having issues with v 2020.4.0 personally.

I'll check again (I've had to have it uninstalled to unblock my work).

**Comment by badgerbaj** [comment]
@peter-sahajian Yes, I have some issues with 2020.4.0 as well, just not as severe. In 2020.4.0, randomly tab complete stops working, as well as inline execution with F8. Reloading the extension clears up those issues though.

**Comment by peter-sahajian** [comment]
@badgerbaj For whatever reason, mine used to completely hang/break on 2020.4.0

Wondering if the dev team has been able to reproduce?

**Comment by SydneyhSmith** [comment]
Unfortunately we still unable to reproduce this--I will mark this as a startup issue

**Comment by graphicsxp** [comment]
I just came across this thread. I have the same issue. I start the debugger in VS Code for a powershell script and the terminal just hands on :  

=====> PowerShell Integrated Console v2020.6.0 <=====

Not sure what do do from there. How can I downgrade the powershell extension to an older version ?

**Comment by SydneyhSmith** [comment]
@graphicsxp thanks for your comment, that actually sounds like a slightly different issue. Could you please open a new issues and include your debug configuration (in addition to the other requests in the issue template), thanks!

**Comment by jdaehler** [comment]
Stuck in process "You have an older version of PackageManagement known to cause issues with the PowerShell extension. Would you like to update PackageManagement (You will need to restart the PowerShell extension after)?"

**Comment by bearkat2173** [comment]
> Stuck in process "You have an older version of PackageManagement known to cause issues with the PowerShell extension. Would you like to update PackageManagement (You will need to restart the PowerShell extension after)?"

I am too

**Comment by SydneyhSmith** [comment]
Thanks @jdaehler and @bearkat2173 that actually sounds like a separate issue from this one, could you please open a new issue? Thanks!

**Comment by GAZelenka** [comment]
Hi, I have been having the stalled start-up problem regarding "PowerShell.Show.SessionMenu failed...", and "older version of PackageManagement known to cause issues..." others have outlined above.

My settings.json file had the powershell.powerShellDefaultVersion set to "Windows Powershell (64)" - by changing the default to one of the defined versions in the settings.json file, and restarting VS Code:
```powershell
"powershell.powerShellDefaultVersion": "PowerShell Core 7.0",
"powershell.powerShellAdditionalExePaths": [
{"exePath": "C:\\Program Files\\PowerShell\\7\\pwsh.exe","versionName": "PowerShell Core 7.0"},
{"exePath": "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe","versionName": "PowerShell Core 5.1"}
]
```
I was able to startup successfully, run code, and access the "Show Session Menu" again.

If I try to set the default back to "Windows Powershell (64)" or "PowerShell Core 5.1" though, the problem recurs.

**Comment by MoonRaker58** [comment]
Seems related to this issue but I would gladly open another issue if needed.

I have been able to replicate the same behavior on:
1. a corporate pc with multiple group policies that I can't view or edit.
2. on 2 personal machines with no group policies and different antivirus clients.

Before documenting here:
1. I removed all powershell extension settings from user settings and from workspace settings.
2. I disabled all extensions except Powershell

Extension version 6.0 and the terminal never progresses past 
`=====> PowerShell Integrated Console v2020.6.0 <=====`
I check the PowerShell extension log and on last line I find
`7/1/2020 12:31:43 PM [ERROR] - <ConsoleFeature>: Unable to instantiate; language client undefined.`

The PIC will launch if I down grade to earlier versions I tried both
`code --install-extension .\PowerShell-Preview-2020.5.0.vsix` or `code --install-extension .\PowerShell-2020.4.0.vsix`

But the extension auto upgrades to 6.0 so on the next reload I'm back to version 6.0 and getting the issue again.
I then tried 6.1 preview and it is the same error behavior as 6.0
Windows restart does not change behavior.

Workaround: I've downgraded to 4.0 and have turned off extension auto updating for now.

**Comment by PetrMa** [comment]
I have the same issue with 2020.6.0. Downgrade to 2020.4.0 and disable auto update solved the issue for now.

**Comment by s12481632** [comment]
Thanks to everyone who commented!  I have the same issue with 2020.6.0. Downgrade to 2020.4.0 and disable auto update solved the issue for now as well.  Ran into the issue on a fresh vscode install with no other extensions.

**Comment by Benny1007** [comment]
I have this same issue. Where do you get the older version to rollback to?

**Comment by MoonRaker58** [comment]
> I have this same issue. Where do you get the older version to rollback to?

See my comments from 15 days ago for a partial set of instructions on how to roll back to older versions.

**Comment by PetrMa** [comment]
> I have this same issue. Where do you get the older version to rollback to?

You can click the Settings image on PowerShell in Extensions and select "Install another version ...." from the menu. Remember that you must also disable Extensions auto update in global settings of VS Code.

**Comment by Liturgist** [comment]
@SydneyhSmith, does the `Needs-Repro-Info` label still apply?

**Comment by jdsanford** [comment]
I am on a corporate systems with many GPOs and McAfee.  I have local administrator access.  When running the current version of the extension the integrated session never finishes starting.  This is the case when running as a user or as an administrator.  Downgrading to 2020.4 resolves the issue.  

Version: 1.47.2 (user setup)
Commit: 17299e413d5590b14ab0340ea477cdd86ff13daf
Date: 2020-07-15T18:22:06.216Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.17763

PSVersion                      6.2.4

**Comment by eoq** [comment]
Same issue with 2020.6.0 and 2020.4.0 works for me as well. I hope this issue is being worked on...
Thanks
Ed

**Comment by asapcloudGIT** [comment]
> es,

I have this issue already for a long time. working with 2020.6.0

**Comment by rjmholt** [comment]
Please try out [this build of the extension](https://dev.azure.com/powershell/_apis/resources/Containers/3076284/vscode-powershell?itemPath=vscode-powershell%2FPowerShell-insiders.vsix) and let us know how it goes.

That build is based on the changes on the linked PR.

**Comment by MoonRaker58** [comment]
@rjmholt I get a not authorized error with that build. I did try preview v2020.7.0 and it works so I'm going to use that one for a while. Thanks for the work.

**Comment by s12481632** [comment]
> @rjmholt I get a not authorized error with that build. I did try preview v2020.7.0 and it works so I'm going to use that one for a while. Thanks for the work.

Same on both counts.  Couldn't access the linked build, but the preview v2020.7.0 build of the extension worked.

**Comment by devedse** [comment]
@rjmholt it seems normal people 😄 don't have access the link you posted:

{"$id":"1","innerException":null,"message":"TF400813: The user 'Windows Live ID\\..........@........' is not authorized to access this resource.","typeName":"Microsoft.TeamFoundation.Framework.Server.UnauthorizedRequestException, Microsoft.TeamFoundation.Framework.Server","typeKey":"UnauthorizedRequestException","errorCode":0,"eventId":3000}

**Comment by TylerLeonhardt** [comment]
Here's the file:

[PowerShell-insiders.vsix.zip](https://github.com/PowerShell/vscode-powershell/files/5071182/PowerShell-insiders.vsix.zip)

Sorry about that

**Comment by devedse** [comment]
@TylerLeonhardt , I verified that this fixes the issue, thanks :)

**Comment by DeanGross** [comment]
2020.6.0 gave me this problem, i unistalled that and then installed 2020.7.0 and the problem still exists.

**Comment by corbob** [comment]
> Here's the file:
> 
> [PowerShell-insiders.vsix.zip](https://github.com/PowerShell/vscode-powershell/files/5071182/PowerShell-insiders.vsix.zip)
> 
> Sorry about that...
> 
> ```
> code --install-extension /full/path/to/PowerShell-insiders.vsix
> ```
> 
> make sure you have the normal PowerShell extension disabled.

@DeanGross  you'll want to use the file Tyler mentioned here. This issue will be resolved in the published release once a new release is pushed.

**Comment by DeanGross** [comment]
@corbob thanks, I have installed v2020.8.0-CI.60063 and still have the problem. I have Windows 10, 1809 OS 17763.1339 if that matters

**Comment by corbob** [comment]
@DeanGross If you're still having the issue, and the CI build didn't solve it, I'd suggest opening a new issue and if possible [attach the logs](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md#providing-information-about-your-environment). It sounds like you might be seeing a different issue than originally reported in this issue.

**Comment by DeanGross** [comment]
@corbob  i updated VSC to 1.49 and 2020.8.0-CI.60063 seems to be working properly now

**Comment by AyJizzy** [comment]
For windows users, when reinstalling vscode I noticed inside properties of the application that 'compatibility mode' was turned on to be be ran with settings for windows 8. Once I removed that it fixed my issue with a restart of the application.

**Comment by kaovilai** [comment]
I had problems after restarting VSCode many times with PowerShell extension v2021.2.2 installed.
The problem when away as of now by restarting Powershell session without restarting VSCode
